### PR TITLE
More E2E tests for Component Browser

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

More cases in E2E tests checking all ways of using component browser.

The tests found one actual bug: sometimes we displayed only a few entries after opening CB because the scroller size was not refreshed (we assumed it will be available in the next tick). Refactored Component Browser so it does not use nextTick anymore.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- ~~[ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - ~~[ ] Unit tests have been written where possible.~~
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui2/src/components/GraphEditor.vue [hunk 1]
```diff
@@ -495,7 +495,12 @@ async function readNodeFromClipboard() {
 
 function handleNodeOutputPortDoubleClick(id: ExprId) {
   componentBrowserUsage.value = { type: 'newNode', sourcePort: id }
-  const placementEnvironment = environmentForNodes([id].values())
+  const srcNode = graphStore.db.getPatternExpressionNodeId(id)
+  if (srcNode == null) {
+    console.log('Impossible happened: Double click on port not belonging to any node: ', id)
```
Comment (somebody1234) [inline app/gui2/src/components/GraphEditor.vue]: `console.error`

File: app/gui2/src/composables/doubleClick.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ export function useDoubleClick<Args extends any[]>(
 
   const handleClick = (...args: Args) => {
     clickCount++
+    console.log(clickCount)
```
Comment (somebody1234) [inline app/gui2/src/composables/doubleClick.ts]: stray log? or maybe hide it behind a debug setting... idk if we have a debug setting though.
we could add a boolean in `util/debug.ts`, or add a define (or multiple) that takes the value from an environment variable (or multiple) so we can do `DEBUG=true npm run dev` or `DEBUG=doubleClick,ls npm run dev`

File: app/gui2/src/components/ComponentBrowser.vue [hunk 3]
```diff
@@ -237,13 +233,18 @@ const selectedSuggestion = computed(() => {
...
+watch(
+  selectedPosition,
+  (newPos) => {
+    if (newPos == null) return
+    highlightPosition.value = newPos
+    if (animatedHighlightHeight.value <= 1.0) {
+      animatedHighlightPosition.skip()
+    }
+  },
+  // Needs to by synchronous to make skipping highlight animation working.
```
Comment (somebody1234) [inline app/gui2/src/components/ComponentBrowser.vue]: `by` -> `be`, `working` -> `work`

File: app/gui2/src/components/ComponentBrowser.vue [hunk 4]
```diff
@@ -21,7 +21,8 @@ import type { Opt } from '@/util/data/opt'
 import { allRanges } from '@/util/data/range'
 import { Vec2 } from '@/util/data/vec2'
 import type { SuggestionId } from 'shared/languageServerTypes/suggestions'
-import { computed, nextTick, onMounted, ref, watch, type ComputedRef, type Ref } from 'vue'
+import { computed, onMounted, ref, watch, type ComputedRef, type Ref } from 'vue'
+import { useScrolling } from './ComponentBrowser/scrolling'
```
Comment (somebody1234) [inline app/gui2/src/components/ComponentBrowser.vue]: prefer `@/components/` rather than `./`

File: app/gui2/e2e/locate.ts [hunk 5]
```diff
@@ -129,9 +129,13 @@ function componentLocator<T extends string>(className: SanitizeClassName<T>) {
 
 export const graphEditor = componentLocator('GraphEditor')
 export const graphNode = componentLocator('GraphNode')
+export function graphNodeByBinding(page: Locator | Page, binding: string) {
+  return graphNode(page).filter({ has: page.locator('.binding').and(page.getByText(binding)) })
+}
 // @ts-expect-error
 export const anyVisualization = componentLocator('GraphVisualization > *')
 export const circularMenu = componentLocator('CircularMenu')
+export const addNewNodeButton = componentLocator('circle-plus')
```
Comment (somebody1234) [inline app/gui2/e2e/locate.ts]: perhaps we should add a `PascalCase` class name matching the component name, for consistency

**Review by somebody1234 — APPROVED** [review]

**Comment by somebody1234** [comment]
i kinda skimmed over the actual E2E tests, someone else may notice if maybe there are some cases that are still not being checked

**Review by Frizi — APPROVED** [review]

**Pull request merged**